principia.props

```diff
@@ -0,0 +1,195 @@
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release_LLVM|x64">
+      <Configuration>Release_LLVM</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|Win32">
+      <Configuration>Release</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|x64">
+      <Configuration>Release</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+  </ItemGroup>
+  
+  <!--Microsoft C++ stuff.-->
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
+
+  <!--Definitions.-->
+  <PropertyGroup>
+    <!--Predicates.-->
+    <PrincipiaCompilerClangLLVM>false</PrincipiaCompilerClangLLVM>
+    <PrincipiaCompilerClangLLVM Condition="$(Configuration)==Release_LLVM">true</PrincipiaCompilerClangLLVM>
+    <PrincipiaOptimize>false</PrincipiaOptimize>
+    <PrincipiaOptimize Condition="$(Configuration)==Release or
+                                  $(Configuration)==Release_LLVM">true</PrincipiaOptimize>
+    <PrincipiaTestProject>true</PrincipiaTestProject>
+    <PrincipiaTestProject  Condition="$(ProjectName) == ksp_plugin or
+                                      $(ProjectName) == serialization or
+                                      $(ProjectName) == benchmarks or
+                                      $(ProjectName) == tools or
+                                      $(ProjectName) == mathematica">false</PrincipiaTestProject>
+    <!--Dependency paths.-->
+    <PrincipiaDependencyConfiguration>Debug</PrincipiaDependencyConfiguration>
+    <PrincipiaDependencyConfiguration Condition="$(PrincipiaOptimize)">Release</PrincipiaDependencyConfiguration>
+  </PropertyGroup>
+
+  <!--Compiler.-->
+  <PropertyGroup>
+    <PlatformToolset>v140</PlatformToolset>
+    <PlatformToolset Condition="$(PrincipiaCompilerClangLLVM)">LLVM-vs2014</PlatformToolset>
+  </PropertyGroup>
+
+  <!--Target.-->
+  <PropertyGroup>
+    <ConfigurationType Condition="$(ProjectName) == serialization">StaticLibrary</ConfigurationType>
+    <ConfigurationType Condition="$(ProjectName) == ksp_plugin">DynamicLibrary</ConfigurationType>
+    <ConfigurationType Condition="$(PrincipiaTestProject) or
+                                  $(ProjectName) == tools or
+                                  $(ProjectName) == benchmarks or
+                                  $(ProjectName) == mathematica">Application</ConfigurationType>
+  </PropertyGroup>
+  <ItemDefinitionGroup Condition="$(ConfigurationType)==Application">
+    <Link>
+      <SubSystem>Console</SubSystem>
+    </Link>
+  </ItemDefinitionGroup>
+
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
+
+  <!--Common options.-->
+  <PropertyGroup>
+    <CharacterSet>Unicode</CharacterSet>
+  </PropertyGroup>
+  <ItemDefinitionGroup>
+    <ClCompile>
+      <AdditionalOptions>/bigobj</AdditionalOptions>
```

Comment:
I think we only need `/bigobj` for `Debug`.

Reply:
Maybe, but let's not add logic for compatibility with pre-Visual C++2005 linkers.